cpp/src/io/csv/datetime.cuh

```diff
@@ -383,3 +383,95 @@ __inline__ __device__ int64_t parseDateTimeFormat(const char *data,
 
   return answer;
 }
+
+// parse integer and update the start position
+template <typename T>
+__inline__ __device__ T parseInteger(const char *data, long &start, long end)
+{
+  bool is_negative = data[start] == '-';
+  T value          = 0;
+
+  long index = start + is_negative;
+  while (index <= end) {
+    if (data[index] >= '0' && data[index] <= '9') {
+      value *= 10;
+      value += data[index] - '0';
+    } else
+      break;
+    ++index;
+  }
+  start = index;
+
+  return is_negative ? -value : value;
+}
+
+__inline__ __device__ bool is_present(
+  const char *data, long &start, long end, const char *needle, int len)
+{
+  if (end - start < len) return false;
+  for (auto i = 0; i < len; i++) {
+    if (data[start + i] != needle[i]) return false;
+  }
+  start += len;
+  return true;
+}
+
+__inline__ __device__ int64_t parseDaysDeltaFormat(const char *data,
+                                                   long start,
+                                                   long end,
+                                                   bool dayfirst)
+{
+  long moving_pos = start;
+  int32_t days    = parseInteger<int>(data, moving_pos, end);
+  return days;
+}
+
+template <typename T>
+__inline__ __device__ int64_t
+parseTimeDeltaFormat(const char *data, long start, long end, bool dayfirst)
+{
+  // %d days [+]%H:%M:%S.n => %d days, %d days [+]%H:%M:%S,  %H:%M:%S.n, %H:%M:%S, %value.
+  int days;
+  int8_t hour, minute, second;
+  int nanosecond     = 0;
+  constexpr char sep = ':';
+
+  // single pass to parse days, hour, minute, seconds, nanosecond
+  long moving_pos = start;
+  int32_t value   = parseInteger<int>(data, moving_pos, end);
+  if (std::is_same<T, cudf::duration_D>::value) return value;
+  while (data[moving_pos] == ' ') moving_pos++;
+  // if (moving_pos >= end) return value;  // %value
+  // " days [+]"
```

Comment:
Should this be removed?

Reply:
comments to describe the code.

Reply:
I enabled it now because that line will be useful for json parsing of timedelta types.